python/ray/dashboard/modules/actor/actor_head.py

```diff
@@ -310,21 +310,13 @@ def _convert_to_dict():
     async def _cleanup_actors(self):
         while True:
             try:
-                if len(DataSource.actors) > MAX_ACTORS_TO_CACHE:
-                    logger.debug("Cleaning up dead actors from GCS")
-                    while len(DataSource.actors) > MAX_ACTORS_TO_CACHE:
-                        if not self.dead_actors_queue:
-                            logger.warning(
-                                f"More than {MAX_ACTORS_TO_CACHE} "
-                                "live actors are cached"
-                            )
-                            break
-                        actor_id = self.dead_actors_queue.popleft()
-                        if actor_id in DataSource.actors:
-                            actor = DataSource.actors.pop(actor_id)
-                            node_id = actor["address"].get("rayletId")
-                            if node_id and node_id != actor_consts.NIL_NODE_ID:
-                                del DataSource.node_actors[node_id][actor_id]
+                while len(self.dead_actors_queue) > MAX_DELETED_ACTORS_TO_CACHE:
+                    actor_id = self.dead_actors_queue.popleft()
```

Comment:
Oh, I found one bug: we should only add actor to `dead_actors_queue` if it's dead and non-restartable